### PR TITLE
G-API: own::Mat size

### PR DESCRIPTION
**Pull request by mpashchenkov**
The same MatSize structure for GAPI's own::Mat

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Magic commands:
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.4.1:20.04
build_image:Custom Win=openvino-2021.4.1
build_image:Custom Mac=openvino-2021.4.1

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by mpashchenkov** [comment]
@rgarnov, it is simple analogue of cv::MatSize for own::Mat.
I have questions:
- about place of MatSize, m b need to move to mat.hpp from types.hpp;
- Does it necessary add deownification for own::MatSize? I think, this struct is more for internal use;
- M B Need to add some specific possibilities for own::Mat or more tests.

Removed extra extensions for own::Size (not theme of this PR I think)

**Review by dmatveev — COMMENTED** [review]
👍 

@rgarnov awaiting for your review here

**Review by dmatveev — APPROVED** [review]

File: modules/gapi/include/opencv2/gapi/own/mat.hpp [hunk 1]
```diff
@@ -239,6 +242,7 @@ namespace cv { namespace gapi { namespace own {
                 tmp.data = tmp.memory.get();
 
                 *this = std::move(tmp);
+                this->size = MatSize(&this->rows, &this->dims);
```
Comment (rgarnov) [inline modules/gapi/include/opencv2/gapi/own/mat.hpp:245]: Looks like there is something incorrect with MatHeader/MatSize operator= if it doesn't work without this line

**Review by rgarnov — CHANGES_REQUESTED** [review]

**Comment by rgarnov** [comment]
> @rgarnov, it is simple analogue of cv::MatSize for own::Mat. I have questions:
> 
> * about place of MatSize, m b need to move to mat.hpp from types.hpp;
> * Does it necessary add deownification for own::MatSize? I think, this struct is more for internal use;
> * M B Need to add some specific possibilities for own::Mat or more tests.
> 
> Removed extra extensions for own::Size (not theme of this PR I think)

It's better to move MatSize to own/mat.hpp since it's used only there
There is no need for deownification of MatSize

**Comment by asmorkalov** [comment]
@mpashchenkov @dmatveev Do you plan to finish the PR?

**Comment by mpashchenkov** [comment]
@asmorkalov, I don't have time for this right now, I'll reopen it later.

**Comment by dmatveev** [comment]
That's sad, that was a good PR actually. @rgarnov would you mind taking this over? You've had comments to address.

**Pull request closed without merging**